src/coprocessor/util.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use kvproto::coprocessor as coppb;
+use tipb::schema::ColumnInfo;
+
+use super::codec::datum::Datum;
+use super::codec::mysql;
+
+pub fn prefix_next(key: &[u8]) -> Vec<u8> {
```

Comment:
Could we add some unit tests in this file?

Reply:
And please improve this function to reduce alloc.

Reply:
@AndreMouche tests added, PTAL